lib/nats.js

```diff
@@ -700,7 +707,22 @@ Client.prototype.setupHandlers = function () {
     return
   }
 
+  let toh
```

Comment:
Why would you set a regular timeout since you have access to `stream`, why not use `stream.setTimeout()` here instead of in `stream.on('connect')`? Would setting the streaming.setTimeout() before "on connect" would not cover the connect itself taking too much time?
If you have to set this non socket-level timer, then I would set it once here and clear it only when the PONG to initial PING has been received.

Reply:
there's no connect timeout, the other timeout there's no handle for it

Reply:
Also, having a timer removed when it is processing the pings, etc, could trigger an unknown amount of time, I believe the correct thing to do is to tighten the check to where it matters.